#! /bin/sh /usr/share/dpatch/dpatch-run
## lossy_comp_test-overflow.dpatch by Martin Michlmayr <tbm@cyrius.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Avoid a possible overflow in sum_abs.
## DP: See #362414.

@DPATCH@
diff -urNad libsndfile-1.0.18~/tests/fix_this.c libsndfile-1.0.18/tests/fix_this.c
--- libsndfile-1.0.18~/tests/fix_this.c	2009-02-07 05:05:20.000000000 +0100
+++ libsndfile-1.0.18/tests/fix_this.c	2009-02-16 09:58:08.000000000 +0100
@@ -69,7 +69,8 @@
 lcomp_test_int (const char *str, const char *filename, int filetype, double margin)
 {	SNDFILE		*file ;
 	SF_INFO		sfinfo ;
-	int			k, m, *orig, *data, sum_abs ;
+	int		k, m, *orig, *data ;
+	long long	sum_abs ;
 	long		datalen, seekpos ;
 	double		scale ;
 
